### ISSUE TITLE
feat(studio): BD3LM block-diffusion streaming via step/stream callbacks

### ISSUE THREAD

**Issue by nishide-dev** [nishide-dev/unturtle#272]
## Background

`_sample_block_diffusion` (introduced as the BD3LM generation path) does not invoke `step_callback` or `stream_callback` at any point. As a result, Studio's SSE progress stream shows nothing during generation and dumps the final text all at once.

`_sample_mdlm` fires these callbacks after each denoising step, giving real-time streaming. BD3LM should match this behaviour.

## Implementation

The two-level loop in `_sample_block_diffusion`:

```
for block in range(num_blocks):          # outer — block-by-block
    for i_step in range(effective_steps): # inner — per-block denoising steps
        ...
        x[:, T_prefix:T_total] = x_block_updated
        # ← fire step_callback and stream_callback here
```

After each inner step update:
- `step_callback(global_step, total_steps)` — update SSE progress counter
- `stream_callback(global_step, total_steps, x.detach().clone())` — stream intermediate token state

Global step = `block_idx * steps_per_block + i_step + 1`. Total = `num_blocks * steps_per_block` (approximate, since `effective_steps` may vary per block).

## Acceptance criteria

- [ ] SSE progress events fire during BD3LM generation (not just at completion)
- [ ] Studio chat shows intermediate token state during BD3LM generation
- [ ] MDLM streaming behaviour is not regressed
- [ ] Unit tests cover callback invocation count and arguments
